### PR TITLE
 Fix issues with parsing invokable macros

### PR DESCRIPTION
**Pull request by standaniels**
## Summary
Fix issues with parsing invokable macros (#1084).

## Type of change
- Bug fix (non-breaking change which fixes an issue)

### Checklist
- [x] Existing tests have been adapted and/or new tests have been added
- [x] Add a CHANGELOG.md entry
- [x] Update the README.md _(n/a)_
- [x] Code style has been fixed via `composer fix-style`

Existing tests pass, but I expect that new tests must be added. Could you help me or point me in the right direction?

### CODE REVIEW

**Comment by KaloyanYosifov** [comment]
@standaniels Would you be able to provide tests for this? I know that there is a another pull request that fixes the same issue  https://github.com/barryvdh/laravel-ide-helper/pull/1123 where @mfn requires a test for this. 

I understand that tests might be tedious, but would really help to know if this functionality is working correctly and to ensure that the functionality doesn't break in future updates.

**Comment by standaniels** [comment]
I've used the `MethodTest` class as a starting point for a new `MacroTest` class. Let me know if there's anything that can be improved.

**Comment by KaloyanYosifov** [comment]
@standaniels Looks really good for me and i think the test covers it pretty much. I used it on one of my big projects and everything seems to be correct. Great job!!!!

The only thing left is, when @mfn has the time to review it.

**Review by barryvdh — APPROVED** [review]

**Review by mfn — APPROVED** [review]
Briefly skimmed it, LGTM

**Comment by KaloyanYosifov** [comment]
@standaniels It seems that there is a conflict from a recent pull request. Would you be able to check it out?

**Comment by standaniels** [comment]
Sure! Let me have a look.

**Comment by standaniels** [comment]
Here you are.

**Comment by mfn** [comment]
Nice, thanks :)

**Comment by KaloyanYosifov** [comment]
@barryvdh @mfn  Is there any problem in merging this?

**Comment by standaniels** [comment]
Thanks for merging. #1084 may now be closed 😊

**Comment by mfn** [comment]
The changelog ended up in the wrong version due to the late merge 😅 Fixed via https://github.com/barryvdh/laravel-ide-helper/pull/1138

**Comment by matsava** [comment]
I'm still seeing this error, is this released?

**Comment by mfn** [comment]
@matsava not yet, it's listed for next release, please see https://github.com/barryvdh/laravel-ide-helper/blob/master/CHANGELOG.md#next-release

File: src/Macro.php [hunk 1]
```diff
@@ -63,7 +63,11 @@ protected function initPhpDoc($method)
 
     protected function addLocationToPhpDoc()
     {
-        $enclosingClass = $this->method->getClosureScopeClass();
+        if ($this->method->name === '__invoke') {
```
Comment (bytestream) [inline src/Macro.php:66]: Given a macro accepts `object|callable`, possible values are:
* `callable` - `function () { }`
* `callable` - `[$instance, 'method']`
* `object` - `new Foo()`

This should be `$enclosingClass === null` so it covers cases all cases where `$this->method` is not an anonymous function.

Comment (mfn) [reply src/Macro.php:66]: Feel free to send a PR!

**Pull request merged**